src/Subscriber.js

```diff
@@ -18,8 +18,31 @@ class Subscriber {
 
     this._retryMap = new Map()
 
-    this._callback = () => Promise.resolve()
     this._initializePromise = undefined
+
+    this.actions = new Map()
+  }
+
+  _callback (msg) {
+    let {action, data} = typeof msg === 'object' ? msg : {}
+    if (!this.actions.has(action)) {
+      return Promise.resolve()
+    }
+
+    let handler = this.actions.get(action)
+    return Promise.resolve(handler(data))
```

Comment:
You can wrap the handler in a promise instead of resolving with its return value:

```suggestion
    return Promise.resolve().then(() => handler.call(this, data))
```

This way the handler can return a promise too.

And also please execute the handler in the context of the class, so users of the library can pass in methods from the class in the constructor like this: `this.registerAction('someAction', this.someActionHandler)`

Reply:
fix